src/BenchmarkDotNet/Toolchains/Mono/MonoToolchain.cs

```diff
@@ -0,0 +1,38 @@
+﻿using BenchmarkDotNet.Toolchains.CsProj;
+using BenchmarkDotNet.Toolchains.DotNetCli;
+using JetBrains.Annotations;
+using System;
+
+namespace BenchmarkDotNet.Toolchains.Mono
+{
+    [PublicAPI]
+    public class MonoToolchain : CsProjCoreToolchain, IEquatable<MonoToolchain>
+    {
+        [PublicAPI] public static readonly IToolchain Mono60 = From(new NetCoreAppSettings("net6.0", null, "mono60"));
+        [PublicAPI] public static readonly IToolchain Mono70 = From(new NetCoreAppSettings("net7.0", null, "mono70"));
+
+        private MonoToolchain(string name, IGenerator generator, IBuilder builder, IExecutor executor, string customDotNetCliPath)
+            : base(name, generator, builder, executor, customDotNetCliPath)
+        {
+        }
+
+        [PublicAPI]
+        public static new IToolchain From(NetCoreAppSettings settings)
+        {
+            var runtimeIdentifier = CustomDotNetCliToolchainBuilder.GetPortableRuntimeIdentifier();
+            return new MonoToolchain(settings.Name,
+                        new MonoGenerator(settings.TargetFrameworkMoniker, settings.CustomDotNetCliPath, settings.PackagesPath, settings.RuntimeFrameworkVersion),
+                        new MonoPublisher(
+                            settings.CustomDotNetCliPath,
+                            $"--self-contained -r {runtimeIdentifier} /p:UseMonoRuntime=true /p:RuntimeIdentifiers={runtimeIdentifier}"),
```

Comment:
I set RuntimeIdentifiers explicitly here because I had the issue describe [here](https://github.com/dotnet/sdk/issues/10566).

Reply:
For that it would be great to add a comment to the code and provide the link to the issue (so in the future we don't remove it by accident).

I also wonder if this logic should be moved to `MonoPublisher`, as it's specific for this publisher and it won't ever be reused by other toolchains.